invenio_app_ils/cli.py

```diff
@@ -7,226 +7,261 @@
-    {
-        "loanid": "loanid-2",
-        "item_pid": "itemid-2",
-        "patron_pid": "1",
-        "state": "PENDING",
-        "transaction_date": "2018-06-29",
-        "transaction_location_pid": "locid-1",
-        "transaction_user_pid": "user_pid",
-        "pickup_location_pid": "locid-1",
-        "request_expire_date": "2018-07-28",
-    },
-    {
-        "loanid": "loanid-3",
-        "item_pid": "itemid-5",
-        "patron_pid": "2",
-        "state": "PENDING",
-        "transaction_date": "2018-06-29",
-        "transaction_location_pid": "locid-1",
-        "transaction_user_pid": "user_pid",
-        "pickup_location_pid": "locid-1",
-        "request_expire_date": "2018-07-28",
-    },
-    {
-        "loanid": "loanid-4",
-        "item_pid": "itemid-7",
-        "patron_pid": "2",
-        "state": "PENDING",
-        "transaction_date": "2018-06-29",
-        "transaction_location_pid": "locid-1",
-        "transaction_user_pid": "user_pid",
-        "pickup_location_pid": "locid-1",
-        "request_expire_date": "2018-07-28",
-    },
-]
-
-
-def _mint_record_pid(pid_type, record):
+from .circulation.receivers import index_item_after_loan_change
+from .pidstore.pids import (  # isort:skip
+    DOCUMENT_PID_TYPE,
+    INTERNAL_LOCATION_PID_TYPE,
+    ITEM_PID_TYPE,
+    LOCATION_PID_TYPE
+)
+from .records.api import Document, Item, Location, InternalLocation
+
+LOCATION = {
+    Location.pid_field: "1",
+    "name": "Central Library",
+    "address": "",
+    "email": ""
+}
+
+ITEM_CIRCULATION_RESTRICTIONS = [None, "FOR_REFERENCE_ONLY"]
+ITEM_MEDIUMS = ["NOT_SPECIFIED", "ONLINE", "PAPER", "CDROM", "DVD", "VHS"]
+ITEM_STATUSES = ["LOANABLE", "MISSING"]
+LOAN_STATUSES = ["PENDING", "ITEM_ON_LOAN", "ITEM_RETURNED", "CANCELLED"]
+
+
+def get_internal_locations(location):
+    """Return random internal locations."""
+    return [
+        {
+            InternalLocation.pid_field: "{}".format(i),
+            Location.pid_field: location[Location.pid_field],
+            "legacy_id": "{}".format(randint(100000, 999999)),
+            "name": "Building {}".format(randint(1, 10)),
+            "notes": lorem.sentence(),
+            "physical_location": lorem.sentence()
+        } for i in range(1, randint(5, 10))
+    ]
+
+
+def get_documents_items(internal_locations):
+    """Return random document and items."""
+    documents = [
+        {
+            Document.pid_field: "{}".format(i),
+            "title": "{}".format(lorem.sentence()),
+            "authors": "{}".format(lorem.sentence()),
+            "abstracts": ["{}".format(lorem.text())],
+        } for i in range(1, randint(10, 20))
+    ]
+    len_docs = len(documents) - 1
+    len_internal_locations = len(internal_locations) - 1
+    LEN_ITEM_CIRCULATION_RESTRICTIONS = len(ITEM_CIRCULATION_RESTRICTIONS) - 1
+    LEN_ITEM_MEDIUMS = len(ITEM_MEDIUMS) - 1
+    LEN_ITEM_STATUSES = len(ITEM_STATUSES) - 1
+
+    items = []
+    for i in range(1, randint(40, 50)):
```

Comment:
I didn't want to parametrize these number as argument of the cli to avoid yet another config, but if you think it will be useful we could have something:
`ils demo data --docs 100 --items 2000 --loans 5000`

Reply:
Added